wordpress.org/public_html/wp-content/plugins/wporg-gp-customizations/inc/cli/class-detect-duplicate-translations.php

```diff
@@ -0,0 +1,286 @@
+						) AS subquery
+					) THEN 'current'
+					ELSE 'old'
+				END,
+				user_id_last_modified = CASE
+					WHEN id = (
+						SELECT id
+						FROM (
+							SELECT id
+							FROM {$wpdb->gp_translations}
+							WHERE original_id = %d
+							AND translation_set_id = %d
+							AND status = 'current'
+							ORDER BY date_modified DESC, id DESC
+							LIMIT 1
+						) AS subquery
+					) THEN user_id_last_modified
+					ELSE NULL
+				END,
+				date_modified = CASE
+					WHEN id = (
+						SELECT id
+						FROM (
+							SELECT id
+							FROM {$wpdb->gp_translations}
+							WHERE original_id = %d
+							AND translation_set_id = %d
+							AND status = 'current'
+							ORDER BY date_modified DESC, id DESC
+							LIMIT 1
+						) AS subquery
+					) THEN date_modified
+					ELSE NOW()
+				END
+				WHERE original_id = %d
+				AND translation_set_id = %d
+				AND status = 'current';
+			",
+				$duplicate['original_id'],
+				$duplicate['translation_set_id'],
+				$duplicate['original_id'],
+				$duplicate['translation_set_id'],
+				$duplicate['original_id'],
+				$duplicate['translation_set_id'],
+				$duplicate['original_id'],
+				$duplicate['translation_set_id']
+			);
+
+			// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+			$wpdb->query( $update_query );
+
+			if ( $wpdb->last_error ) {
+				// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+				echo 'Error in the SQL query: ' . $wpdb->last_error;
+				$this->last_sql_error = $wpdb->last_error;
+				break;
+			}
+
+			if ( $this->print_sql ) {
+				// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+				echo "- Translation_set_id: {$duplicate['translation_set_id']}, Original_id: {$duplicate['original_id']} -> ";
+				// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+				echo $update_query . "\n";
+			}
+		}
+
+		if ( $this->verbose ) {
+			// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+			echo 'Number of translations updated: ' . number_format_i18n( count( $this->duplicates ) ) . "\n";
+		}
+	}
+
+	/**
+	 * Notify the duplicates in Slack.
+	 *
+	 * @return void
+	 */
+	private function notify_duplicates() {
+		$mysql_error = '';
+		if ( $this->last_sql_error ) {
+			$mysql_error = "The last MySQL error was: " . $this->last_sql_error . "\n";
+		}
+
+		if ( 0 == count( $this->duplicates ) ) {
+			$message = ":white_check_mark: There are no duplicate translations in the database with *current* status.\n";
+			$message .= $mysql_error;
+		} else {
+			$message  = ":warning: There are " . count( $this->duplicates ) . " duplicate translations in the database with *current* status.\n\n";
+			$message .= "Execute *'wp wporg-translate detect-duplicate-translations --url=translate.wordpress.org --verbose'* to show the duplicate entries.\n";
+			$message .= "Execute *'wp wporg-translate detect-duplicate-translations --url=translate.wordpress.org --update-values'* to update the duplicate entries, solving the problem.\n";
```

Comment:
It irks me a bit wrong that the `detect-duplicate-translations` modifies data with the right parameter. What if we renamed it to `duplicate-translations` and the `verbose` parameter to `detect` and the `update-values` to `fix`?
```suggestion
			$message .= "Execute *'wp wporg-translate duplicate-translations --url=translate.wordpress.org --detect'* to show the duplicate entries.\n";
			$message .= "Execute *'wp wporg-translate duplicate-translations --url=translate.wordpress.org --fix'* to update the duplicate entries, solving the problem.\n";
```

Reply:
I changed the command name and a parameter name:
- Command name: detect-duplicate-translations -> duplicate-translations
- Parameter name: --update -> --fix

By default, the command, without any flag, will detect the duplicated values. The "--verbose" flag will show some important values, like the number of duplicates, and the "--print-sql" flag will show the SQL queries to show and to update the duplicate entries.